### ISSUE TITLE
CSS for list style types

### ISSUE THREAD

**Issue by rbeezer** [rbeezer/mathbook#829]
Every list outputted as HTML has
```HTML
style="list-style-type: xxxx; ..."
```
This appears to be a CSS property that EPUB will interpret, so should we make CSS classes for these?  There would be nine, I think.  Maybe cleaner code on my end if I make up class names that mirror the PTX names for the options (e.g. roman, arabic, etc. for enumerated lists).  I'll drop the necessary info here and build the changes on a branch, given a general go-ahead, or suggestions.  Part of #732.

If I recall correctly, CSS is limited in its ability to do much with this aspect of the look of a list.

https://www.w3.org/TR/CSS21/generate.html#propdef-list-style-type

https://www.w3.org/TR/css-lists-3/

**Comment by davidfarmer** [comment]
It makes sense to try class names that follow the PTX names.

There are some tricky ways to customize markers,
but we don't do that because of accessibility issues.

Let's get a working example and then we can ask people to take
a look.

**Comment by rbeezer** [comment]
Example at:

http:///mathbook.pugetsound.edu/beta/20180203-list-marker/lists.html

The nine values for `list-style-type` in use with PTX have been appended with `-marker` to make nine new class names.  These values are

For `ol`
```
decimal
upper-alpha
lower-alpha
upper-roman
lower-roman
```

For `ul`
```
disc
square
circle
none
```

So, for example, new `ol.decimal-marker` formerly had an inline `style="list-style-type: decimal;"`.

**Comment by davidfarmer** [comment]
I implemented all of those, I think.

What about eliminating the "-marker" part of the class?

Currently the selector I use is, for example,
    .disc-marker

I could have used
    ul.disc-marker

If we eliminate the -marker, then it would be a choice between

    .disc

and

    ul.disc

Not yet obvious to me which is better.

Is non-marker allowed for ol?

**Comment by rbeezer** [comment]
I would prefer to drop the `-marker`.  I thought that perhaps without it, then it would look like another `time.time` situation.  I've got no good input on the selector you might use.

PTX had a "none" option for an `ol`.  I decided that was silly.  debatable, I guess.

An ordered list should have items that are ordered, like steps in a recipe, not the ingedients.  And if ordered, then markers from an ordered set are...in order?  That's the way i see it.

I'll make a new sample, likely after breakfast.

**Comment by rbeezer** [comment]
[Test-link](http:///mathbook.pugetsound.edu/beta/20180203-list-marker/lists.html), as above, updated

No more `-marker`, except for the `none` style I used `ul.no-marker`.

**Comment by rbeezer** [comment]
Alright, a bad thing - it appears.  Entries of multicolumn lists have been receiving inline styles with hard-coded lists.  I suspect these snuck back in while refactoring knowl production.  I've removed them on this branch, presuming the `colsN` attributes on the enclosing `ol, ul` would then take effect.  Not so with updated test link.

See 12.2 and 12.3 which should each hold a two-column list of four items.

**Comment by davidfarmer** [comment]
.colsN should be working, but test article only seems to
have cols2 .

Source of 12.2 seems to have switched from numbers to letters.
Test link did not include Section 5.2, but that .cols5 should be
okay.

I think I didn't mess up legacy multicolumn examples, but it is
hard to be sure.  Details: I actually over-ride part of the inline
style, because of an unrelated padding-left I changed a while
ago.  Seems okay in the places I looked.
